src/insights.jsx

```diff
@@ -237,7 +254,23 @@ function show_connect_dialog() {
                     caption: _("Connect"),
                     style: "primary",
                     clicked: (update_progress) => {
-                        return PK.install_missing_packages(install_data, update_install_progress(update_progress)).then(() => register(update_progress));
+                        return PK.install_missing_packages(install_data, update_install_progress(update_progress)).then(() =>
+                            new Promise((resolve, reject) => {
+                                register(update_progress)
+                                        .then(() => resolve())
+                                        .catch((err, data) => {
+                                            let msg = spawn_error_to_string(err, data);
```

Comment:
I guess that `catch_error` should be used here too. When registration is terminated with errors, then all errors are printed on one line. I tested this with custom insights-client:

```bash
#!/bin/bash

echo "Error to stdout"
>&2 echo "Error to stderr"

exit 1
```

The error message is displayed like this:

![image](https://user-images.githubusercontent.com/2057012/167814253-7efc341b-8f7d-44c9-bdbf-a81292598cb2.png)

Reply:
As I wrote, unfortunately none of my attempts to make the multiline text working there worked. See the longer explanation on why.